### DISCUSSION TITLE
Exception in ASGI application TypeError: 'str' object is not callable

### DISCUSSION THREAD

**Discussion by itzlayz** [fastapi/fastapi#10540] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request, templating, APIRouter
from fastapi.staticfiles import StaticFiles
from fastapi.middleware.cors import CORSMiddleware

from telethon import TelegramClient, types, errors
from uvicorn import Config, Server

import os
import sys
import atexit
import asyncio
import configparser

class MainWeb:
    def __init__(self):
        self.api = FastAPI()
        self.config = Config(self.api, host='0.0.0.0', port=8000)
        self.server = Server(self.config)
        
        self.login_data = {
            'id': 123,
            'hash': "123",
            'phone': None,
            'phone_hash': None,
            'phone_code': None,
            '2fa': None
        }
        self.client = TelegramClient('./teagram',
                                     self.login_data['id'],
                                     self.login_data['hash'])
        
        self.api.add_middleware(
            CORSMiddleware,
            allow_origins=["*"],
            allow_methods=["*"],
            allow_headers=["*"],
            allow_credentials=True,
        )
        self.api.mount(
            "/static", 
            StaticFiles(directory="teagram/web/static"), 
            "static"
        )
        self.api.add_route('/', self.index, methods=["GET"])
        self.api.add_route('/tokens', self.loginToClient, methods=["POST"])

    async def index(self, request: Request):
        return templating.Jinja2Templates(
            directory='teagram/web/templates/'
        ).TemplateResponse(
            'index.html', {'request': request}
        )

    async def loginToClient(self, data: Request):
        try:
            _data = data.headers
            
            if not getattr(self.login_data, 'client', None):
                self.client = TelegramClient('./teagram', _data['id'], _data['hash'])
                await self.client.connect()

                self.login_data['self.client'] = self.client

            config = configparser.ConfigParser()
            try:
                _id = config.get('telethon', 'api_id')
                _hash = config.get('telethon', 'api_hash')
            except:
                _id = None
                _hash = None

                if not (_data['id'] and _data['hash']) and not (_id and _hash):
                    return 'Enter all api tokens'
                else:
                    config["telethon"] = {
                        "api_id": _data['id'],
                        "api_hash": _data['hash']
                    }
                    
                    with open("./config.ini", "w") as file:
                        config.write(file)

            self.login_data['id'] = _data['id'] or _id
            self.login_data['hash'] = _data['hash'] or _hash
            
            try:
                if await self.client.get_me():
                    shutdown()
                    await self.client.disconnect()
                    try:
                        asyncio.get_running_loop().stop()
                    except:
                        pass
                else:
                    await self.client.disconnect()
                    return 'qrcode'
            except Exception as error:
                await self.client.disconnect()
                return error
        except:
            await self.client.disconnect()
            await self.check_tokens(data)
```

### Description

make post request to /token

INFO:     127.0.0.1:42416 - "POST /tokens HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/itzlayz/.local/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/itzlayz/.local/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 91, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/cors.py", line 146, in simple_response
    await self.app(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/itzlayz/.local/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/itzlayz/.local/lib/python3.10/site-packages/starlette/routing.py", line 69, in app
    await response(scope, receive, send)
TypeError: 'str' object is not callable

### Operating System

Windows

### Operating System Details

windows wsl 2

### FastAPI Version

0.103.1

### Pydantic Version

2.2.1

### Python Version

Python 3.11.4

**Comment by YuriiMotov** [comment]
Please, create [MRE](https://stackoverflow.com/help/minimal-reproducible-example) using as few dependencies as possible.

Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
